app/controllers/planning_application/consultations_controller.rb

```diff
@@ -81,5 +90,19 @@ def assign_params
         @attributes = consultation_params
       end
     end
+
+    def update_letter_statuses
+      return if @consultation.neighbour_letters.none?
+
+      # This is not ideal for now as will block the page loading, if it becomes a problem this would be
+      # a good place for optimisation
+      @consultation.neighbour_letters.each do |letter|
```

Comment:
Evangelos mentioned a background job for the letter sending; perhaps that would work here too (in a later PR).

Or alternatively (or additionally?) perhaps just doing it through AJAX and updating the status asynchronously that way. Either way this is fine for now, of course.

Reply:
Yeah I would just worry about the status changing while the people were looking at the page and then them missing it. Or them looking at the page and it showing an incorrect status because the background job is out of date. 

Basically I kind of want to see how much of a problem this is before committing either way

Reply:
Yes, that's true. We'd want to show some sort of indication rather than just showing an old status and silently replacing it. (A loading icon instead of the status at first perhaps?) It might also work to run all of the fetches in parallel within the controller method, with threads.

But absolutely, yeah, if it turns out we can fetch dozens of statuses without significantly impacting the page load then we shouldn't worry about it.